src/components/MapboxMap.tsx

```diff
@@ -1,21 +1,28 @@
 import { School } from "@/types/school";
 import mapboxgl, { LngLatBounds } from "mapbox-gl";
-import { useEffect, useRef } from "react";
+import { useEffect, useRef, useState, useMemo } from "react";
+import { useMapContext } from '@/contexts/MapContext';
 
 type MapboxMapProps = {
-  setSelectedSchool: (school: School | false | null) => void;
+  setSelectedSchool: (school: School | null) => void;
   selectedSchool: School | false | null;
   schools: School[];
 };
 
-const MapboxMap = ({
-  setSelectedSchool,
-  selectedSchool,
-  schools,
-}: MapboxMapProps) => {
+const MapboxMap = ({ schools }: MapboxMapProps) => {
+  const { selectedSchool, setSelectedSchool } = useMapContext();
   const mapContainer = useRef<HTMLDivElement | null>(null);
   const mapRef = useRef<mapboxgl.Map | null>(null);
   const markersRef = useRef<{ [key: string]: mapboxgl.Marker }>({});
+  const [mapLoaded, setMapLoaded] = useState(false);
+  const userHasInteracted = useRef(false);
+
+  const flyToOptions = useMemo(() => ({
+    zoom: 14, // zoom level to fly to
```

Comment:
I'm noticing that we are always zooming. I think it's best not to change zoom unless the user explicitly requests it.

Reply:
Yeah @nickvisut, it can get zoomy, happy to change anything. I already worked on the settings to slow down the zoom effect to make it less dizzying,. But we can remove the auto-zoom completely. Design team didn't ask for any changes, but maybe they don't know how easy it is to change, not sure.

Reply:
maybe with 100+ schools it might actually make more sense to auto-zoom as the pins will be right on top of each other. i still can't imagine so many pins in my mind's eye, it is going to look awfully crowded.

Reply:
Agreed that that could be a future problem, but the experience as of now is suboptimal. Consider: A user arrives at the map and clicks on a school. How do they then click on another school? They will have to manually zoom out in most cases.

There may be a good solution that is the best of both worlds, but let's lean towards best UX w/current set of schools while we figure that out.